### ISSUE TITLE
Quote Lockdown

### ISSUE THREAD

**Issue by nickbe** [satrun77/tinyissue#60]
We've started using quotes as a simple controlling tool.
For this to work there's one thing we could use:

Managers and Admins should have an option to mark a quote as:
OPEN / LOCKED / SETTLED.

>Once a developer has entered his time estimation on the managers request and they may have discussed this, the manager should be able to lock the quote and prevent further modification.

>Additionally the admin/manager might have to do something with the quote later on. Like invoicing or whatever to settle the quote.

So I'd say tickets should be visible somewhere in a list of "quoted and unsettled" tickets.
Never forget to invoice a customer when you did some work! Good idea?

**Comment by nickbe** [comment]
Just just discussed this here:

We think a simple "lock" switch would be best. 

- Only managers (and admins) should see it.
- User shouldn't see quotes or informations about this at all
- Developers should only see the quote information when it's locked. Otherwise they can edit the estimation.

Everything else can be solved with statuses.

**Comment by nickbe** [comment]
I found some bugs here when testing:

1. When editing the issue again after having locked the estimation then the lock flag is not shown as set. So when saving the issue then the estimation gets accidentially unlocked.
2. I think as a developer I should see the estimated time even when it's locked.
3. When I changed the status as a developer and tried to save I got this error:
ErrorException in CrudTrait.php line 107:
Undefined index: time_quote

**Comment by nickbe** [comment]
Thanks and closed :)

**Comment by nickbe** [comment]
Strange... the quote lockdown seems to have vanished somewhere on the way.
I cannot lock the quote anymore.

**Comment by satrun77** [comment]
I have downloaded the database from the debug instance (thanks for the link). The problem is that the database is out of sync with the upgrade scripts. 

To fix open the sqlite database, then`ti_migrations` table and delete the record about `2016_06_25_090714_add_quote_lock_to_issue` then run the migration command. This will fix the database structure.

I managed to replicate the issue with the downloaded sqlite and fix it with above steps
